iml-agent/src/action_plugins/ldev.rs

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+use iml_wire_types::LdevEntry;
+use std::collections::BTreeSet;
+use tokio::{
+    fs::{metadata, read_to_string, File},
+    io::AsyncWriteExt,
+};
+
+async fn write_to_file(content: String) -> Result<(), ImlAgentError> {
+    let ldev_path = env::get_ldev_conf();
+    let mut file = File::create(ldev_path).await?;
+    file.write_all(content.as_bytes()).await?;
+
+    Ok(())
+}
+
+async fn read_ldev_config() -> Result<String, ImlAgentError> {
+    let ldev_path = env::get_ldev_conf();
+
+    match metadata(&ldev_path).await {
+        Ok(_) => Ok(read_to_string(&ldev_path).await?),
+        Err(_) => Ok("".into()),
+    }
+}
+
+fn parse_entries(ldev_config: String) -> BTreeSet<LdevEntry> {
+    ldev_config.lines().map(LdevEntry::from).collect()
+}
+
+fn config_needs_update_check(
```

Comment:
nit: is this fn even worth having? It's just doing a `!=` check at this point.

Reply:
I could see both sides but if you're questioning it then it's probably an unnecessary abstraction. I'll take it out.